treescript/src/treescript/merges.py

```diff
@@ -0,0 +1,147 @@
+"""Treescript merge day functionality."""
+import logging
+import os
+import shutil
+
+import attr
+
+from scriptworker_client.utils import makedirs
+from treescript.mercurial import get_revision, run_hg_command
+from treescript.task import get_merge_config
+from treescript.versionmanip import do_bump_version, get_version
+
+log = logging.getLogger(__name__)
+
+
+def replace(file_name, from_, to_):
+    """Replace text in a file."""
+    log.info("Replacing %s -> %s inside %s", from_, to_, file_name)
+    with open(file_name) as f:
+        text = f.read()
+    new_text = text.replace(from_, to_)
+    if text == new_text:
+        raise ValueError(f"{file_name} does not contain {from_}")
+    with open(file_name, "w") as f:
+        f.write(new_text)
+
+
+def touch_clobber_file(repo_path):
+    """Update the clobber file in the root of the repo."""
+    log.info("Touching clobber file")
+    clobber_file = os.path.join(repo_path, "CLOBBER")
+    with open(clobber_file) as f:
+        contents = f.read()
+    new_contents = ""
+    for line in contents.splitlines():
+        line = line.strip()
+        if line.startswith("#") or line == "":
+            new_contents += f"{line}\n"
+    new_contents += "Merge day clobber"
+    with open(clobber_file, "w") as f:
+        f.write(new_contents)
+
+
+async def apply_rebranding(config, repo_path, merge_config):
+    """Apply changes to repo required for merge/rebranding."""
+    log.info("Rebranding %s to %s", merge_config.get("from_branch"), merge_config.get("to_branch"))
+    if merge_config.get("version_files"):
+        current_version = get_version("browser/config/version.txt", repo_path)
+        next_version = f"{current_version.major_number}.{current_version.minor_number}"
+
+        await do_bump_version(config, repo_path, merge_config["version_files"], next_version)
+    if merge_config.get("version_files_suffix"):
+        current_version = get_version("browser/config/version.txt", repo_path)
+        current_version = attr.evolve(current_version, is_esr=False, beta_number=None, is_nightly=False)
+        next_version = f"{current_version}{merge_config.get('version_suffix')}"
+
+        await do_bump_version(config, repo_path, merge_config["version_files_suffix"], next_version)
+
+    for f in merge_config.get("copy_files", list()):
+        shutil.copyfile(os.path.join(repo_path, f[0]), os.path.join(repo_path, f[1]))
+
+    for f, from_, to in merge_config.get("replacements", list()):
+        replace(os.path.join(repo_path, f), from_, to)
+
+    touch_clobber_file(repo_path)
```

Comment:
Is it possible to add a parameter in the task description to skip running "touch_clobber_file()"? Thunderbird repos won't have that.

Reply:
This PR isn't intended for Thunderbird work - it doesn't cover all of the manual Firefox tasks yet. Could we add those when we add the Thunderbird tasks?

Reply:
Of course.

Reply:
Thank you for being understanding!